common/main/java/com/couchbase/lite/ProxyAuthenticator.java

```diff
@@ -71,7 +73,10 @@ public boolean equals(Object o) {
         if (this == o) { return true; }
         if (!(o instanceof ProxyAuthenticator)) { return false; }
         final ProxyAuthenticator other = (ProxyAuthenticator) o;
-        return username.equals(other.username) && Arrays.equals(password, other.password);
+        return username.equals(other.username)
+            && MessageDigest.isEqual(
+            new String(password).getBytes(StandardCharsets.UTF_8),
+            new String(other.password).getBytes(StandardCharsets.UTF_8));
```

Comment:
Is `Arrays.equals(password, other.password)` not safe?

Reply:
I've never heard about it, before... but this: https://jira.issues.couchbase.com/browse/CCV-239